src/src/com/microsoft/aad/adal/UnexpectedServerResponseException.java

```diff
@@ -0,0 +1,14 @@
+package com.microsoft.aad.adal;
+
+import java.io.IOException;
+
+/**
+ * Created by imakhal on 2/17/2016.
+ */
+public class UnexpectedServerResponseException extends IOException {
```

Comment:
I just noticed that this exception is being used for at least two different types of servers and their responses.  For the device challenge response I think we need a separate exception that is also derived from AuthenticationException.  I would call it AuthenticationServerProtocolException.  In a subsequent PR all OAuth and other protocol level errors should use this new exception.